From fa45d0d9c9109f3c2953ad02d7e3aa69c4e09add Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 1 Dec 2014 11:50:00 +0100
Subject: [PATCH] base/init.sh: ignore exclamation mark for kmod static-nodes

newer versions add an exclamation mark for the type, to mark it to be
created on boot only.
---
 modules.d/99base/init.sh | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/modules.d/99base/init.sh b/modules.d/99base/init.sh
index 3afe6d5..5f66d1f 100755
--- a/modules.d/99base/init.sh
+++ b/modules.d/99base/init.sh
@@ -73,15 +73,16 @@ fi
 if command -v kmod >/dev/null 2>/dev/null; then
     kmod static-nodes --format=tmpfiles 2>/dev/null | \
         while read type file mode a a a majmin; do
-        case $type in
-            d)
-                mkdir -m $mode -p $file
-                ;;
-            c)
-                mknod -m $mode $file $type ${majmin%:*} ${majmin#*:}
-                ;;
-        esac
-    done
+            type=${type%\!}
+            case $type in
+                d)
+                    mkdir -m $mode -p $file
+                    ;;
+                c)
+                    mknod -m $mode $file $type ${majmin%:*} ${majmin#*:}
+                    ;;
+            esac
+        done
 fi
 
 trap "action_on_fail Signal caught!" 0
